src/db/singleton.py

```
@@ -0,0 +1,14 @@
+import threading
+
+
+class SingletonMeta(type):
+    _instances = {}
```

Ruff (0.14.5)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)